src/eins-hwinfo.c

```diff
@@ -105,159 +115,70 @@ round_to_nearest (guint64 size,
   return (guint32) CLAMP (size / divisor, 0, G_MAXUINT32);
 }
 
-GVariant *
-eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-                                          GError **error)
+guint32
+eins_hwinfo_get_ram_size (void)
+{
+  glibtop_mem mem;
+  guint32 size_mib;
+
+  glibtop_get_mem (&mem);
+  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
+  return size_mib;
+}
+
+gboolean
+eins_hwinfo_get_disk_space_for_partition (GFile          *file,
+                                          DiskSpaceType  *diskspace)
 {
   g_autoptr(GFileInfo) info = NULL;
+  g_autoptr(GError) error = NULL;
   guint64 total = 0;
   guint64 used = 0;
-  guint64 available = 0;
+  guint64 free = 0;
+
+  if (diskspace == NULL)
+    {
+      g_error ("Wrong memory address to hold disk space information");
+      return FALSE;
+    }
+
+  diskspace->total = 0;
+  diskspace->used = 0;
+  diskspace->free = 0;
 
   info = g_file_query_filesystem_info (file,
                                        G_FILE_ATTRIBUTE_FILESYSTEM_SIZE ","
                                        G_FILE_ATTRIBUTE_FILESYSTEM_USED ","
                                        G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
                                        NULL /* cancellable */,
-                                       error);
+                                       &error);
   if (info == NULL)
     {
-      /* TODO: g_file_peek_path() from GLib 2.56. */
-      g_autofree gchar *path = g_file_get_path (file);
-      g_prefix_error (error, "Couldn't get disk space for %s: ", path);
-      return NULL;
+      const gchar *path = g_file_peek_path (file);
+      g_prefix_error (&error, "Couldn't get disk space for %s: ", path);
```

Comment:
If you're not going to return `error` to the caller there is no point in prefixing it – nothing will ever show the error.

```suggestion
      g_warning ("Couldn't get disk space for %s: %s", path, error->message);
```

Reply:
Personally I would prefer the `GError **` out argument, which the caller can either supply (and log warning), or ignore.

Reply:
ok